website/source/docs/vagrant-cloud/api.html.md

```diff
@@ -0,0 +1,493 @@
+##### **402** Payment Required
+
+You are trying to access a resource which is delinquent on billing.
+Please contact the owner of the resource so that they can update their billing information.
+
+##### **403** Forbidden
+
+You are attempting to use the system in a way which is not allowed.
+There could be required request parameters that are missing, or one of the parameters is invalid.
+Please check the response `errors` key, and double-check the examples below for any discrepancies.
+
+##### **404** Not Found
+
+The resource you are trying to access does not exist. This may also be returned if you attempt to access a private resource that you don't have authorization to view.
+
+##### **429** Too Many Requests
+
+You are currently being rate-limited. Please decrease your frequency of usage, or contact us at [support+vagrantcloud@hashicorp.com](mailto:support+vagrantcloud@hashicorp.com) with a description of your use case so that we can consider creating an exception.
+
+#### Server Errors
+
+##### **500** Internal Server Error
+
+The server failed to respond to the request for an unknown reason.
+Please contact [support+vagrantcloud@hashicorp.com](mailto:support+vagrantcloud@hashicorp.com) with a description of the problem so that we can investigate.
+
+##### **503** Service Unavailable
+
+Vagrant Cloud is temporarily in maintenance mode.
+Please check the [HashiCorp Status Site](http://status.hashicorp.com) for more information.
+
+
+## Authentication Tokens
+
+### Create a token
+
+`POST /api/v1/authenticate`
+
+Creates a new token for the given user.
+
+#### Arguments
+
+* `token`
+    * `desription` (Optional) - A description of the token.
+* `user`
+    * `login` - Username or email address of the user authenticating.
+    * `password` - The user's password.
+
+#### Example Request
+
+<div class="examples">
+  <ul class="examples-header">
+    <li class="examples-menu examples-menu-shell"><a onclick="setExampleLanguage('shell');">cURL</a></li>
+    <li class="examples-menu examples-menu-ruby"><a onclick="setExampleLanguage('ruby');">Ruby</a></li>
+  </ul>
+  <div class="examples-body">
+    ```shell
+    curl \
+      --header "Content-Type: application/json" \
+      https://app.vagrantup.com/api/v1/authenticate \
+      --data '
+        {
+          "token": {
+            "description": "Login from cURL"
+          },
+          "user": {
+            "login": "me",
+            "password": "secret"
+          }
+        }
+      '
+    ```
+
+    ```ruby
+    # gem install http, or add `gem "http"` to your Gemfile
+    require "http"
+
+    api = HTTP.persistent("https://app.vagrantup.com").headers(
+      "Content-Type" => "application/json"
+    )
+
+    response = api.post("/api/v1/authenticate", json: {
+      token: { description: "Login from Ruby" },
+      user: { login: "me", password: "secret" }
+    })
+
+    if response.status.success?
+      # Success, the response attributes are available here.
+      p response.parse
```

Comment:
Is `p` here something special (I assume it's printing...)? I don't see any require other than `http`.

Reply:
`p` is built-in to Ruby, and is the equivalent of `puts object.inspect`. I would prefer `pp` here, but `p` doesn't have to be require'd.

Bigger picture though, I'm open to any suggestions that help someone get started with the response without pushing them in a certain direction.

Reply:
Oh, I didn't realize `p` was built in! That makes sense. For some reason I thought it was something like `pp` but also an external library. 👍